scripts/debug/collect-logs.sh

```diff
@@ -0,0 +1,81 @@
+
+CLUSTER_NAME=${CLUSTER_NAME:-"test-$(date +%s)"}
+ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"
+
+# dump logs from kind and all the nodes
+dump-logs() {
+  # log version information
+  echo "=== versions ==="
+  echo "kind : $(kind version)" || true
+  echo "kubectl: "
+  kubectl --kubeconfig=$CLUSTER_NAME.kubeconfig version || true
+  echo ""
+
+  # dump all the info from the CAPI related CRDs
+  mkdir -p $ARTIFACTS/logs
+  kubectl get \
+  clusters,azureclusters,machines,azuremachines,kubeadmconfigs,machinedeployments,azuremachinetemplates,kubeadmconfigtemplates,machinesets,kubeadmcontrolplanes \
+  --all-namespaces -o yaml >> "${ARTIFACTS}/logs/capz.info" || true
+
+  # dump images info
+  echo "images in docker" >> "${ARTIFACTS}/logs/images.info"
```

Comment:
consider using a pattern like the following:
```shell
# dump images info
 {
   echo "images in docker"
   docker images
   echo "images from bootstrap using containerd CLI"
   docker exec kind-control-plane ctr -n k8s.io images list
 } >> "${ARTIFACTS}/logs/images.info"
```

Reply:
so fresh, so clean